appeals/web/src/server/appeals/appeal-documents/appeal-documents.controller.js

```diff
@@ -635,6 +637,100 @@ export const postUploadDocumentVersionCheckAndConfirm = async ({
  */
 
 /** @typedef {import('@pins/appeals.api').Schema.Document} Document */
+/**
+ * @param {Object} params
+ * @param {import('@pins/express/types/express.js').Request} params.request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} params.response
+ * @param {string} params.backButtonUrl
+ */
+export const renderChangeDocumentFilename = async ({ request, response, backButtonUrl }) => {
+	const {
+		currentFolder,
+		errors,
+		params: { appealId, documentId }
+	} = request;
+
+	if (!currentFolder) {
+		return response.status(500).render('app/500.njk');
+	}
+
+	const currentFile = await getFileInfo(request.apiClient, appealId, documentId);
+
+	if (!currentFile) {
+		return response.status(500).render('app/500.njk');
+	}
+
+	const mappedPageContent = ChangeDocumentFilenamePage(backButtonUrl, currentFolder, currentFile);
+
+	return response.status(200).render('appeals/documents/add-document-details.njk', {
+		pageContent: mappedPageContent,
+		errors
+	});
+};
+
+/**
+ * @param {Object} params
+ * @param {import('@pins/express/types/express.js').Request} params.request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} params.response
+ * @param {string} params.backButtonUrl
+ * @param {string} [params.nextPageUrl]
+ */
+export const postChangeDocumentFilename = async ({
+	request,
+	response,
+	backButtonUrl,
+	nextPageUrl
+}) => {
+	try {
+		const {
+			body,
+			apiClient,
+			params: { appealId },
+			errors
+		} = request;
+
+		if (errors) {
+			return await renderChangeDocumentFilename({ request, response, backButtonUrl });
+		}
+
+		const currentFile = await getFileInfo(request.apiClient, appealId, body.documentId);
+
+		if (!currentFile) {
+			return response.status(500).render('app/500.njk');
+		}
+
+		const { dateReceived, redactionStatus } = currentFile.latestDocumentVersion;
+
+		const redactionStatuses = await getDocumentRedactionStatuses(apiClient);
+
+		const apiRequest = mapDocumentFilenameFormDataToAPIRequest(
+			body,
+			dateReceived,
+			// @ts-ignore
```

Comment:
What error is being ignored here? It would be good to resolve that if possible

Reply:
Type resolved and ts-ignore removed